R/ResampleResult.R

```diff
@@ -80,4 +82,128 @@ fortify.ResampleResult = function(model, data, measure = NULL, ...) {
     variable.name = "measure_id", value.name = "performance")
 }
 
+plot_learner_prediction_resample_result = function(learner, object, measure, grid_points = 100L, expand_range = 0) {
+  UseMethod("plot_learner_prediction_resample_result")
+}
+
+plot_learner_prediction_resample_result.LearnerClassif = function(learner, object, measure, grid_points = 100L, expand_range = 0) {
+
+  task = object$task
+  features = task$feature_names
+  learners = object$learners
+
+  if (length(features) != 2) {
+    stop("plot_learner_prediction.TaskClassif only works for tasks with two features!")
+  }
+
+  grid_res = predict_grid(learners, task, grid_points = grid_points, expand_range = expand_range)
+
+  if (length(learners) > 1) {
+    facet_labels = setNames(paste(object$resampling$id, seq_len(object$resampling$iters)), seq_len(object$resampling$iters))
+    folds_facet = facet_wrap(".id", labeller = as_labeller(facet_labels))
+  } else {
+    folds_facet = NULL
+  }
+
+  if (learner$predict_type == "prob") {
+    grid_res[, "prob.response" := .SD[, paste0("prob.", get("response")), with = FALSE] , by = "response"]
+    raster_aes = aes_string(fill = "response", alpha = "prob.response")
+  } else {
+    raster_aes = aes_string(fill = "response")
+  }
+
+  ggplot(grid_res, aes_string(features[1L], features[2L])) +
+    geom_raster(raster_aes) +
+    geom_point(data = task_data(object), aes_string(features[1L], features[2L], fill = task$target_names, shape = ".point_type"), size = 4, color = "black") +
+    scale_shape_manual(values = c(train = 21, test = 22, both = 23), name = "set") +
+    folds_facet
+}
+
+plot_learner_prediction_resample_result.LearnerRegr = function(learner, object, measure, grid_points = 100L, expand_range = 0) {
+
+  task = object$task
+  features = task$feature_names
+  learners = object$learners
+
+  if (length(features) > 2) {
+    stop("plot_learner_prediction.LearnerRegr only works with one or two features!")
+  }
+
+  grid = predict_grid(learners, task, grid_points, expand_range)
+
+  if (length(learners) > 1) {
+    facet_labels = setNames(paste(object$resampling$id, seq_len(object$resampling$iters)), seq_len(object$resampling$iters))
+    folds_facet = facet_wrap(".id", labeller = as_labeller(facet_labels))
+  } else {
+    folds_facet = NULL
+  }
```

Comment:
This is basically copy-paste.

Reply:
There probably is more. Not cleaned up yet.